### PR TITLE
Keychain: Add documentation

### PR DESCRIPTION
**Pull request by mikolasstuchlik**
Documentation for keychain property wrapper alongside some minor changes.

Partially implements #19

### CODE REVIEW

File: Sources/FTPropertyWrappers/Keychain - Support/KeychainItemPropertyWrapper.swift [hunk 1]
```diff
@@ -1,18 +1,31 @@
 import Foundation
 
+/// This class does not override all requirred properties, using this class as such will result in fatalError. Properties `itemClass` and `primaryKey` needs to be overriden.  Common property wrapper components based on services provided in `SingleValueKeychainItem`.  This class is responsible for providing bridge between binary-based keychain API and typed system. It is also responsible for synchronizing data and deciding which query should be executed.
```
Comment (mkj-is) [inline Sources/FTPropertyWrappers/Keychain - Support/KeychainItemPropertyWrapper.swift]:
```suggestion
/// This class does not override all requirred properties, using this class as such will result in fatalError. Properties `itemClass` and `primaryKey` needs to be overriden. Common property wrapper components based on services provided in `SingleValueKeychainItem`. This class is responsible for providing bridge between binary-based keychain API and typed system. It is also responsible for synchronizing data and deciding which query should be executed.
```

File: Sources/FTPropertyWrappers/Keychain - Support/SingleValueKeychainItem.swift [hunk 2]
```diff
@@ -1,54 +1,90 @@
 import Foundation
 
+/// `SingleValueKeychainItem` is an "abstract" base class for keychain items that are differentiable using certain combination of it's attributes as a form of primary key. This class does not have any safety check when more than one item corresponds to the primary key.
+///
+/// This class provides following services to it's subclasses.:
+///
+/// *Query composition and execution.* Queries are composed from three main sources of data. 1) `QueryElement` property wrappers. Data from wrapped values and it's metadata (like keys and constraints) are collected and composed into a query. When fetch query was executed, those properties are updated accordingly. Refer to `QueryElement` for more implementation details. 2) `itemClass` property. This propety's value is used as a value for key `kSecClass`. 3) `primaryKey` property contains list of kSecAttr**** identifiers which should be excluded from updates and are used to identify the keychain item in fetch, update and delte queries. *Query value and fetch resukts* are read and passed to `itemData` computed property.
```
Comment (mkj-is) [inline Sources/FTPropertyWrappers/Keychain - Support/SingleValueKeychainItem.swift]: Feel free to use markdown formatting to make it more readable. (In this case numbered lists.)

Screenshot 2020-02-27 at 10 07 55
```suggestion
*Query composition and execution.* Queries are composed from three main sources of data.
///
/// 1. `QueryElement` property wrappers. Data from wrapped values and it's metadata (like keys and constraints) are collected and composed into a query. When fetch query was executed, those properties are updated accordingly. Refer to `QueryElement` for more implementation details.
/// 2. `itemClass` property. This propety's value is used as a value for key `kSecClass`.
/// 3. `primaryKey` property contains list of kSecAttr**** identifiers which should be excluded from updates and are used to identify the keychain item in fetch, update and delte queries. *Query value and fetch resukts* are read and passed to `itemData` computed property.
```

File: Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift [hunk 3]
```diff
@@ -1,45 +1,103 @@
...
+/// keychain response. It provides user with easy-to-read API that describes relation between a value and a key. It
+/// also specifies what constraints or access permissions do apply. When request into keychain is being prepared or
+/// results are being parsed, properties of this type are going to be scanned by mirror reflecting enclosing class
+/// instance. At this time, wrapped properties and constrains are being read or written to. Notice, that if you have
+/// more than one `QueryElement` in hiearchy with the same key, behavior may be undefined. In such a case
+/// only guarantee is, that if at least one of such properties is not nil, this key will be written to query before contraints
+/// are resolved.
 public final class QueryElement<T>: WrappedConfiguringElement {
 
+    /// Value corresponding to the attribute key. *Notice: when a value of corresponding key is alredy present in
```
Comment (mkj-is) [inline Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift]: Missed notice.

File: Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift [hunk 4]
```diff
@@ -1,45 +1,103 @@
...
     let constraints: [KeychainQueryPresenceConstraint]
+
+    /// Type anonymous setter and getter for propety stored into or loaded from keychain query.
     var wrappedAsAnonymous: Any? {
         get { wrappedValue }
         set { wrappedValue = newValue.flatMap {$0 as? T} }
     }
 
+    /// Creates regular query element. Elements created with this initializer are written to queries and parsed from
+    /// responses. *Notice: when a value of corresponding key is alredy present in the keychain, setting wrapped
```
Comment (mkj-is) [inline Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift]: Another missed notice.

File: Sources/FTPropertyWrappers/Keychain - Support/KeychainItemPropertyWrapper.swift [hunk 5]
```diff
@@ -1,18 +1,46 @@
...
+/// *Coding.* This class uses module-internal coder which ensures, that all "single value" values
+/// could be ancoded (with exception for floating point types). For compatibility purposes, if `T` is
+/// `Data`, coder does no coding or decoding. This behavior is implemented this way in order to enable
+/// user to access data, that are not encoded in a particular way, for example .utf16 encoded string,
+/// ultra wide integers or JSON strings.
+///
+/// *Synchronization and query execution.* This class can be set to either refresh data when read
+/// or write operation is executed on `wrappedProperty` (notice, read-werite operation like += 1
+/// on Int will result in fetch and update operation at the same time) or manual operation execution using
+/// `try saveToKeychain()`, `try loadFromKeychain()` and `try deleteKeychain()`
```
Comment (mikolasstuchlik) [inline Sources/FTPropertyWrappers/Keychain - Support/KeychainItemPropertyWrapper.swift]:
```suggestion
/// `try saveToKeychain()`, `try loadFromKeychain()` and `try deleteKeychain()`.
```

File: Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift [hunk 6]
```diff
@@ -1,45 +1,103 @@
...
+/// also specifies what constraints or access permissions do apply. When request into keychain is being prepared or
+/// results are being parsed, properties of this type are going to be scanned by mirror reflecting enclosing class
+/// instance. At this time, wrapped properties and constrains are being read or written to. Notice, that if you have
+/// more than one `QueryElement` in hiearchy with the same key, behavior may be undefined. In such a case
+/// only guarantee is, that if at least one of such properties is not nil, this key will be written to query before contraints
+/// are resolved.
 public final class QueryElement<T>: WrappedConfiguringElement {
 
+    /// Value corresponding to the attribute key. *Notice: when a value of corresponding key is alredy present in
+    /// the keychain, setting wrapped value to nil will not unset this attribute in keychain.*
```
Comment (mikolasstuchlik) [inline Sources/FTPropertyWrappers/Keychain - Support/KeychainTypes.swift]:
```suggestion
    /// the keychain, setting wrapped value to nil will not unset this attribute in keychain.
```

**Pull request merged**